test/con_cache_test.exs

```diff
@@ -184,6 +184,24 @@ defmodule ConCacheTest do
     end)
   end
 
+  test "get_or_store on bag" do
+    with_cache([ets_options: [:bag]],fn(cache) ->
+      assert ConCache.get_or_store(cache, :a, fn() -> 1 end) == 1
```

Comment:
I'm not 100% sure what should we do here, but I think the current implementation is definitely not good.

The first call to `ConCache.get_or_store(cache, :a, fn() -> 1 end)` will return `1`, while all subsequent calls will return `[1]`. That's inconsistent, so we should fix it.

Since currently `get_or_store` is key-based, my proposal was to allow the store to insert multiple values for bag types. The downside of this is that lambda has to return a list for bags, and a single element for sets. As you point out, this means different semantics for different types, which is somewhat confusing. It's not a precedent though, since e.g. `get` works in the same way (it returns a value, or a list of values, depending on the table type).

Another option is to disallow `get_or_store`  for bags. Perhaps that's the sanest option. Implementing it in the client code is easy, and atomicity can easily be obtained with [isolated/4](https://hexdocs.pm/con_cache/ConCache.html#isolated/4). So perhaps that's a better option.

What do you think?